support/upsert/upsert.go

```diff
@@ -21,6 +24,12 @@ type CreateOrUpdateProvider interface {
 	CreateOrUpdate(ctx context.Context, c crclient.Client, obj crclient.Object, f controllerutil.MutateFn) (controllerutil.OperationResult, error)
 }
 
+var withStatusSubresource = sets.NewString(fmt.Sprintf("%T", &capiawsv1beta1.AWSCluster{}))
```

Comment:
Would a gvk make sense here? or is the list of types we'll support so limited that we don't care about name collisions?

Reply:
I think packagename.Type is likely to remain unique, especially since the CAPI resources have the providername in the typename (i.E. its AWSCluster, not Cluster). Should I end up being wrong about this its easy to change later on